src/http/client/builder.rs

```diff
@@ -535,13 +542,42 @@ impl<T> EasyHttpConnectorBuilder<T, HttpStage> {
     ///
     /// Use `wait_for_pool_timeout` to limit how long we wait for the pool to give us a connection
     ///
+    /// This also applies a [`RequestVersionAdapter`] layer to make sure that request versions
+    /// are adapted when pooled connections are used, which you almost always need, but in case
+    /// that is unwanted, you can use [`Self::with_custom_connection_pool_without_version_adapter`] instead.
```

Comment:
Would  it not make sense to drop the `without_version_adapter` variants and just assume that people can do all that using the existing `with_custom_connection_pool`?

Reply:
I think it's a good idea that you add `RequestVersionAdapter` to the setters where we use the default pool. I think however that we do not need the `without_request_version_adapter` and neither do we need it for the custom pool. I think you are at a point where you have the knowledge to implement and use your own pool that you also can be responsible for knowing if you need the adapter or not. Especially if we add a big fat warning to the docs of the custom pool setter.